types/jquery/v2/index.d.ts

```diff
@@ -37,7 +37,7 @@ MERCHANTABLITY OR NON-INFRINGEMENT.
 See the Apache Version 2.0 License for specific language governing permissions
 and limitations under the License.
 ***************************************************************************** */
-
+// TypeScript Version: 3.2
```

Comment:
I don't see any reason this comment needs to be added to begin with, as none of these changes depend on TS 3.2. I would recommend removing this line.

Reply:
```
Error in jquery/v2
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/jquery/v2/jquery-tests.ts:3544:14
ERROR: 3544:14  expect  Compile error in typescript@3.1 but not in typescript@3.2.
Fix with a comment '// TypeScript Version: 3.2' just under the header.
Expected at least 1 arguments, but got 0.
```

Ah I thought somebody told me to add it there... So, should I add it again? Or is there another workaround?

Reply:
If the only error is coming from `jquery-tests.ts`, perhaps you are using a feature in the test file that requires 3.2, even though the definitions themselves do not?

Reply:
IIRC, the only thing that could cause this difference would be the change that "void parameters are considered optional". Considering that `void` is a core conceit of this PR, that may be intentional; however I can't see us bumping the minimum required TS version for something as widely used as `jquery` this high quite yet.

I imagine this can be "fixed" in old TS versions by adding a noop type parameter to the zero-argument overload of `$.Deferred`, this way is can be selected on older TS versions when `void` (or anything else) is supplied as a type argument.

Reply:
Do you mean something like this:

```
Deferred(): JQueryDeferred<JQueryStatic["noop"]>;
Deferred<T>(beforeStart?: (deferred: JQueryDeferred<T>) => any): JQueryDeferred<T>;
```

Reply:
By "noop type parameter", I mean a normal type parameter on the signature itself (like the second overload) which is simply unused. (With accompanying lint suppression and ideally a default)